nixos/modules/virtualisation/lxd.nix

```diff
@@ -93,11 +93,15 @@ in
 
     security.apparmor = {
       enable = true;
-      profiles = [
-        "${cfg.lxcPackage}/etc/apparmor.d/usr.bin.lxc-start"
-        "${cfg.lxcPackage}/etc/apparmor.d/lxc-containers"
-      ];
       packages = [ cfg.lxcPackage ];
+      policies = {
+        "bin/lxc-start".profile = ''
+          #include ${cfg.lxcPackage}/etc/apparmor.d/usr.bin.lxc-start
+        '';
+        "lxc-containers".profile = ''
+          #include ${cfg.lxcPackage}/etc/apparmor.d/lxc-containers
```

Comment:
Why have files containing only a single include directive each, rather than copying the files to be included (e.g., `….profile = builtins.readFile "${cfg.lxcPackage}/…"`)?  (I think in the latter case store optimisation should deduplicate the files, and, last I heard (which was years ago), it's easier on e.g. ZFS to have fewer tiny files.)

Reply:
For the sake of traceability, and because not all filesystems dedup and not all machines have store optimisation turned on, I'm against that idea

Reply:
Okay.  By _traceability_, do you mean the ability to tell that the profile is (or is equivalent to) a profile provided by upstream, or hasn't been (problematically) modified?  Why does copying versus include-ing make much of a difference?

Reply:
I mean it makes it easy to answer the question "what package did this text come from" without having to dig through nixpkgs code. Otherwise, for saving on what amounts to a handful of small files, it seems like a weird overcomplication of things

Reply:
I no longer meant to promote copying the file, but I remained confused, but I think I've answered my question by finding that, given a `/etc/apparmor.d/bin/lxc-start` that looks like—

```
include /nix/store/n8cvxn9sy84pz3s4vmdva8hylwkd92lg-lxc-4.0.3/etc/apparmor.d/usr.bin.lxc-start
```

<!--Certainly, this would suggest that profile ultimately being used (by which I mean the profile being `include`d) *may* be from the `nixos.lxc` package (although so does the filename `/etc/apparmor.d/bin/lxc-start`).  Given that, I could try to answer the question "what package did this text come from" by checking whether `nix-build '<nixpkgs>' -A lxc` does print "/nix/store/n8cvxn9sy84pz3s4vmdva8hylwkd92lg-lxc-4.0.3", but then, if I can do that, I can also check-->

—there is the command `nix-store -qb pname /nix/store/n8cvxn9sy84pz3s4vmdva8hylwkd92lg-lxc-4.0.3` that will say that the profile came from a derivation with `pname = "lxc"`, by which one then could determine that, _if_ it came from nixpkgs, then it came from `pkgs.lxc`, which possibility one then could check by building `pkgs.lxc` and seeing that it has the same store path hash.

At the time of my initial comment, I'm afraid I was, for which I apologize, unaware of any way to determine that the profile came from `pkgs.lxc` besides making the educated (and likely harder-to-automate) guess that it did and checking the store path hash of `pkgs.lxc`, in which case it wouldn't be much more difficult to check that a copied file equals some file from `pkgs.lxc`.